frontend/ui/elements/font_settings.lua

```diff
@@ -5,7 +5,9 @@ local _ = require("gettext")
 
 --[[ Font settings for systems with multiple font dirs ]]--
 
-local LINUX_FONT_PATH = "share/fonts"
+local LINUX_FONT_PATH = os.getenv("XDG_DATA_HOME")
+    and string.format("%s/fonts", os.getenv("XDG_DATA_HOME"))
+    or string.format("%s/%s", Device.home_dir, ".local/share/fonts")
```

Comment:
`string.format()` is probably a leftover from LuaJIT 2.0, back when it didn't JIT compile regular concat with `..`.

That is, it's probably more easily legible if you simply do:
```suggestion
    or Device.home_dir .. "/.local/share/fonts")
```

Reply:
I see, that looks neat! I'm really new with lua, so I didn't know about that, let me refactor this a bit

Reply:
It's a performance optimization, one that's presumably fairly irrelevant in this particular context, but it was good as a default habit at the time. The difference is that `string.format` was implemented in C/asm and `..` was interpreted. Depending on context `table.concat` might still be preferable, but generally not unless you're dealing with large strings.

Cf. <http://wiki.luajit.org/NYI> as to why you might see some patterns in LuaJIT that you wouldn't generally see in Lua. If it says `2.1` instead of `yes` that means it's something we would've mildly avoided in the past.

NB The LuaJIT interpreter is written in [DynASM](https://luajit.org/dynasm.html) which is typically still 2-4 × faster than vanilla Lua.

Reply:
Thanks for the info!